index.js

```diff
@@ -356,11 +387,14 @@ export function signRecoverable(h, d, e) {
   if (!isExtraData(e)) {
     throw new Error(THROW_BAD_EXTRA_DATA);
   }
-  const [signature, recoveryId] = necc.signSync(h, d, { der: false, extraEntropy: e, recovered: true });
-  return { signature, recoveryId }
+  const sig = secp256k1.sign(h, d, { extraEntropy: e });
+  return {
+    signature: sig.toCompactRawBytes(),
+    recoveryId: sig.recovery,
+  };
 }
 
-export function signSchnorr(h, d, e = Buffer.alloc(32, 0x00)) {
+export function signSchnorr(h, d, e = new Uint8Array(32)) {
```

Comment:
Why are you filling `e` with zeros by default?? This massively decreases signature security and is against bip340 spec which mandates RANDOM e

Reply:
This behavior was implemented to match `@bitcoinjs-lib/tiny-secp256k1`, as the aim of `bitcoinerlab/secp256k1` is to be a drop-in replacement for it. Users previously raised concerns about [[differences between the libraries](https://github.com/bitcoinerlab/secp256k1/issues/3)](https://github.com/bitcoinerlab/secp256k1/issues/3).

If you recall, we discussed this topic in the past: https://github.com/paulmillr/noble-curves/issues/61

`bitcoinjs-lib/tiny-secp256k1` is aligned with `bitcoin-core/secp256k1`, which also exhibits this behavior. However, since `tiny-secp256k1` has recently switched to the Rust `secp256k1` crate, perhaps this has been updated. Could you confirm, @junderw?

Reply:
Ok.

to clarify, libsecp doesn’t use it because they can’t access good randomness. We can. Sipa mentioned:

> Using random nonces is recommended and obviously better, but libsecp256k1 has no access to high-quality randomness; it's up to the caller to provide it.